src/compile/layer.ts

```diff
@@ -207,9 +213,9 @@ export class LayerModel extends Model {
 
   public assembleMarks(): any[] {
     // only children have marks
-    return flatten(this.children.map((child) => {
+    return assembleLayerMarks(this, flatten(this.children.map((child) => {
```

Comment:
Can we just rename this `assembleLayerSelectionMarks`?   A little more verbose, but clearer.

I think the old comment `// only children have marks` might not make sense any more too.